internal/plans/objchange/plan_valid.go

```diff
@@ -447,14 +447,18 @@ func assertPlannedObjectValid(schema *configschema.Object, prior, config, planne
 		}
 
 	case configschema.NestingSet:
-		if !planned.IsKnown() || !config.IsKnown() {
-			// if either is unknown we cannot check the lengths
+		plannedL := planned.Length()
+		configL := config.Length()
+
+		// if the length is not known, check the value range
+		if !configL.IsKnown() || !plannedL.IsKnown() {
+			if ok := plannedL.Range().Includes(configL); ok.IsKnown() && ok.False() {
+				errs = append(errs, path.NewErrorf("count in plan (%#v) disagrees with count in config (%#v)", plannedL, configL))
+			}
```

Comment:
I think it should be correct to _just_ do this range-based check, because calling `Range()` on a known number will return a range whose upper and lower bound is the exact value, and so `Includes(...)` will effectively just do an equality check in that case.

(This range API is intentionally designed to minimize the need for lots of finicky special cases for unknown values, so if that _doesn't_ work for some reason then I'd like to know about it so I can think about how to improve this abstraction in future.)

Reply:
oh, that makes sense! in that case I'll simplify this a bit more!